apps/system/js/advanced_telemetry.js

```diff
@@ -512,6 +512,82 @@
     return JSON.stringify(packet);
   };
 
+  // This method converts a raw histogram with the .counts and .ranges arrays
+  // in it to one with just .values using the dict format.
+  AdvancedTelemetryPing.prototype.packHistogram =
+  function packHistogram(hgram) {
+    //Constant for Histogram types
+    const HISTOGRAM_EXP = 0;
+
+    if (typeof hgram.ranges === 'undefined' ||
+      typeof hgram.counts === 'undefined') {
+      return hgram;
+    }
+    let r = hgram.ranges;
+    let c = hgram.counts;
```

Comment:
is there any way that `ranges` or `counts` could be an empty list rather than just an undefined value?